opentelemetry-api/src/opentelemetry/util/_decorator.py

```diff
@@ -73,7 +73,7 @@ async def async_wrapper(*args: Pargs, **kwargs: Pkwargs) -> R:
 
 def _agnosticcontextmanager(
```

Comment:
```suggestion
def _agnosticcontextmanager(
    func: "Callable[P, Iterator[R]]",
) -> "Callable[P, _AgnosticContextManager[R]]":
    @functools.wraps(func)
    def helper(*args: Pargs, **kwargs: Pkwargs) -> _AgnosticContextManager[R]:
        return _AgnosticContextManager(func, args, kwargs)

    # Ingoring the type to keep the original signature of the function
    return helper  # type: ignore[return-value]
```

Hello,

Sorry for the delay.

After checking locally with `mypy`, I propose to ignore the return type as it's the same:

- `P`: Parameter specification variable defined by `ParamSpecArgs(self)` and `ParamSpecKwargs(self)` (https://github.com/python/cpython/blob/3.10/Lib/typing.py#L867)
- `[VarArg(Pargs), KwArg(Pkwargs)]` : Define the `*args: Pargs, **kwargs: Pkwargs` passed as arguments of the function (https://github.com/open-telemetry/opentelemetry-python/pull/3795/files#diff-2e55e8b8bf4ad1efcb8000b54cb4f7c2d22a04292fce425714204a9ebb0db4b9L78)

Here is some code I use to test if the typing is correctly forwarded to the child functions :

```python
from opentelemetry.util._decorator import _agnosticcontextmanager
from opentelemetry.trace import get_tracer
import typing


@_agnosticcontextmanager
def manager(bla: str) -> typing.Iterator[int]:
    yield len(bla)


@manager(bla="hello")
def hello() -> str:
    return "world"


with manager(bla="hello") as m:
    print(m)

print(hello())

tracer = get_tracer(__name__)

with tracer.start_as_current_span("test") as span:
    print(span)

@tracer.start_as_current_span("test")
def test() -> bool:
    return True

print(test())
```

IMO if your LSP is giving you the right types at the end, you can move forward :)